lib/sensors/temperature.cpp

```diff
@@ -0,0 +1,76 @@
+#include "temperature.hpp"
+
+namespace hyped::sensors {
+
+Temperature::Temperature(const std::uint8_t mux_address,
+                         const std::uint8_t channel,
+                         hyped::io::I2c &i2c,
+                         hyped::core::ILogger &log)
+    : log_(log),
+      i2c_(i2c),
+      channel_(channel),
+      mux_address_(mux_address)
+{
+}
+
+Temperature::~Temperature()
+{
+}
+
+std::optional<std::uint16_t> Temperature::read()
+{
+  const auto status_check_result = i2c_.readByte(mux_address_, kStatus);
+
+  const auto temperature_high = i2c_.readByte(mux_address_, kData_T_H);
+  const auto temperature_low  = i2c_.readByte(mux_address_, kData_T_L);
+
+  // getting the temperature_high_byte and temperature_low_byte values from the register
+
+  if (!status_check_result) {
+    log_.log(hyped::core::LogLevel::kFatal, "Temperature could not read status");
+    return std::nullopt;
+  }
+
+  if (status_check_result.value() == 0) {
+    log_.log(hyped::core::LogLevel::kFatal, "Temperature sensor is not ready to be read from");
+    return std::nullopt;
+  }
+
+  if (!temperature_high) {
+    log_.log(hyped::core::LogLevel::kFatal, "Temperature high could not be read");
+    return std::nullopt;
+  }
+
+  if (!temperature_low) {
+    log_.log(hyped::core::LogLevel::kFatal, "Temperature low could not be read");
+    return std::nullopt;
+  }
+
+  std::uint16_t temperature = (temperature_high.value() << 8) | temperature_low.value();
```

Comment:
think we shud keep this to `int16_t` to preserve sign, jic its negative due to a faulty sensor

Reply:
also make this `const` and directly return `temperature` scaled appropriately